mimesis/providers/personal.py

```diff
@@ -272,23 +273,25 @@ def email(self, domains: Union[tuple, list] = None) -> str:
             domain=domain,
         )
 
-    def social_media_profile(self) -> str:
+    def social_media_profile(self, site: Optional[SocialNetwork] = None) -> str:
         """Generate profile for random social network.
 
         :return: Profile in some network.
 
         :Example:
             http://facebook.com/some_user
         """
+        if site is None:
+            site = SocialNetwork.get_random_item()
 
-        # TODO: enum here
-        urls = [
-            'facebook.com/{}',
-            'twitter.com/{}',
-            'medium.com/@{}',
-        ]
-        url = 'http://' + self.random.choice(urls)
-        username = self.username(template='U_d')
+        if site in SocialNetwork:
+            site = SOCIAL_NETWORKS[site.value]
+        else:
+            raise NonEnumerableError(SocialNetwork)
+
+        url = 'https://www.' + site
+        template = self.random.choice(['l-d', 'l_d', 'Ud', 'ld'])
```

Comment:
templates are doubled, you already have them somewhere else

Reply:
@sobolevn I'll update it.